### PR TITLE
Certificates domains

### PR DESCRIPTION
**Pull request by ejayesh**
initial working code for domain bootstrap

### CODE REVIEW

**Review by lgarvey — APPROVED** [review]
Just some comments.  Not sure now important it is to fix things now as we'll revisit this script post PoC. So happy to just merge, if you're OK

File: domain-bootstrap.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+            #breakpoint()
+            print(records['Name'], " found")
+            records_to_add.append(records)
+            add_records(client, records, subdom_id)
+
+    #breakpoint()
+    #print("Records added: ", records_to_add)
+
+
+    return
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply domain-bootstrap.py:191]: ok

File: domain-bootstrap.py [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+def check_r53(domain_profile, project_profile, domain, base_domain):
+    # route 53 / ACM POC
+    pp = pprint.PrettyPrinter(depth=6)
+
+
+    PROJECT_ACC_PROFILE = project_profile
+    DOMAIN_ACC_PROFILE = domain_profile
+
+    project_session = boto3.session.Session(profile_name=PROJECT_ACC_PROFILE)
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply domain-bootstrap.py:277]: ok

File: domain-bootstrap.py [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+    print(f"Logged in with AWS Project account: {account_name[0]}/{sts_proj.get_caller_identity()['Account']}\nUser: {sts_proj.get_caller_identity()['UserId']}")
+    #breakpoint()
+    # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/acm.html
+    # create the certificate
+
+    #breakpoint()
+    response = domain_client.list_hosted_zones_by_name()
+
+    # check if prod, feature not implemented yet.
+    LIVE = False
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply domain-bootstrap.py:300]: ok

File: domain-bootstrap.py [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+            # We only want to go 2 sub domains deep in R53
+            if (len(parts) - base_len) < MAX_DOMAIN_DEPTH:
+                create_hosted_zone(domain_client, domain, subdom, base_domain)
+
+            break
+
+        parts.pop(0)
+    else:
+        print(f"No hosted zone found for {domain}")
+        return
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply domain-bootstrap.py:331]: exit(1) sounds good

**Comment by ejayesh** [comment]
yes, lets merge for now and improve the code post PoC

**Pull request merged**